server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -524,10 +546,30 @@ public TSExecuteStatementResp executeQueryStatement(TSExecuteStatementReq req) {
    *             AuthorPlan
    */
   private TSExecuteStatementResp internalExecuteQueryStatement(String statement,
-      long statementId, PhysicalPlan plan, int fetchSize, String username) {
+      long statementId, PhysicalPlan plan, int fetchSize, String username) throws IOException {
     auditLogger.info("Session {} execute Query: {}", currSessionId.get(), statement);
     long startTime = System.currentTimeMillis();
     long queryId = -1;
+    if (plan instanceof QueryPlan && config.isEnablePerformanceTracing()) {
+      File performanceDir = SystemFileFactory.INSTANCE.getFile(config.getPerformanceDir());
```

Comment:
put all these performance related to a class TracingManager, which manage a writer, log file.

Reply:
Fixed.